.github/workflows/on-push-to-main.yml

```diff
@@ -0,0 +1,192 @@
+name: On push to main
+
+on:
+  push:
+    branches: [main]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+env:
+  SIMPLETEST_DB: "mysql://drupal:drupal@127.0.0.1:3306/drupal"
+  SIMPLETEST_BASE_URL: "http://127.0.0.1:8080"
+  MODULE_FOLDER: "drupal-cache"
+  MODULE_REPO: "momentohq/drupal-cache"
+  DRUPAL_MODULE_NAME: "momento_cache"
+  DRUPAL_CORE_VERSION: 9.4.x
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+  # MOMENTO_API_TOKEN: ${{ secrets.ALPHA_TEST_AUTH_TOKEN }}
+  MOMENTO_API_TOKEN: ${{ secrets.MOMENTO_API_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      fail-fast: true
+      matrix:
+        php-version: ['7.4', '8.0', '8.1'] # , '8.2']
```

Comment:
PHP 8.2 is incompatible with Drupal 9.4, which is what we're targeting right now. I'll expand testing to PHP 8.2 and Drupal 9.5 and 10 when I can.